api/models/app.go

```diff
@@ -520,30 +537,53 @@ func (a *App) RunAttached(process, command, releaseId string, height, width int,
 					}
 				}
 
-				if *container.Name == process && releaseMatch {
+				if *releaseContainer.Name == process && releaseMatch {
 					break ContainerCheck
 				}
-				container = nil
+				releaseContainer = nil
 			}
+			////////////////////////////////////////////////////
 
-			if container != nil {
+			if releaseContainer != nil {
+				container = releaseContainer
 				break
 			}
 		}
-	}
 
-	if container == nil {
-		return fmt.Errorf("unable to find contaier for %s", process)
+		// If container is nil, the release most likely hasn't been promoted and thus no TaskDefinition for it.
+		if releaseContainer == nil {
+			fmt.Printf("Unable to find contaier for %s. Basing container off of most recent release: %s.\n", process, a.Release)
+			unpromotedRelease = true
```

Comment:
I was thinking the same before, perhaps with using the dynamo table as a reference but nothing stood out at me. 

Thought about using the created timestamp and comparing that to the active timestamp. Wasn't sure if it was reliable enough.